fix(gatsby-plugin-feed): fix link not being replaced and expose custom_namespaces

geocine opened the pull request:
## Description

`link` is not properly being replaced because the underlying option used by `node-rss` is `site_url` so I just created a mapping. `custom_namespaces` is another option used in `node-rss` to specify custom namespaces.

### Documentation

`link` is already in documentation. added `custom_namespaces` to the documentation.

### Tests

I clone a copy of this plugin using this process to test https://www.gatsbyjs.com/docs/creating-a-local-plugin/

## Related Issues

LekoArts commented on packages/gatsby-plugin-feed/src/gatsby-node.js:
```diff
@@ -32,6 +32,7 @@ exports.onPostBuild = async ({ graphql, reporter }, pluginOptions) => {
     const { setup, ...locals } = {
       ...options,
       ...feed,
+      site_url: feed.link,
```
This is already provided through the `site_url: siteUrl` inside the GraphQL query.

To be fair, the docs don't highlight this requirement enough and an addition to the README would be much appreciated!

As such, please revert this change as it would introduce a breaking change (the `link` is optional and thus with this change it would break the plugin for a lot of people)

LekoArts replied:
No, `link` is the the `siteUrl`, see: https://validator.w3.org/feed/docs/rss2.html

> link | The URL to the HTML website corresponding to the channel.

LekoArts commented:
I removed your addition to the plugin options because it's already covered by this `unknown` here:
https://github.com/gatsbyjs/gatsby/blob/0e9feebf11d071fb912ad25649b068eb39b78d8b/packages/gatsby-plugin-feed/src/plugin-options.js#L13

I updated the README though, let me know if this is clearer to you.

geocine commented:
> I removed your addition to the plugin options because it's already covered by this `unknown` here:
> 
> https://github.com/gatsbyjs/gatsby/blob/0e9feebf11d071fb912ad25649b068eb39b78d8b/packages/gatsby-plugin-feed/src/plugin-options.js#L13
> 
> I updated the README though, let me know if this is clearer to you.

Agree I just didn't want the warning showing up. 

> I removed your addition to the plugin options because it's already covered by this `unknown` here:
> 
> https://github.com/gatsbyjs/gatsby/blob/0e9feebf11d071fb912ad25649b068eb39b78d8b/packages/gatsby-plugin-feed/src/plugin-options.js#L13
> 
> I updated the README though, let me know if this is clearer to you.

Thank you, that make its clearer. I added `custom_namespaces` because I didn't want to see warnings.

LekoArts commented:
I've tried it locally with an example site of mine, when I use e.g. `language` inside `feeds` I don't get a warning.
